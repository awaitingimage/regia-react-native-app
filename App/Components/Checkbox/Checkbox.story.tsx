import { storiesOf } from "@storybook/react-native";
import * as React from "react";

import Checkbox from "./Checkbox";

storiesOf("Checkbox")
.add("Default", () => (
  <Checkbox
    text="Get started"
  />
));
