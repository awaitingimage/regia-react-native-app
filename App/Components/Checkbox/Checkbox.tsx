import * as React from "react";
import { Switch, Text, TextStyle, TouchableOpacity, View } from "react-native";
import ExamplesRegistry from "../../Services/ExamplesRegistry";
import styles from "./Style";

interface Props {
  onPress: (checked: boolean) => void;
  text?: string;
  checked?: boolean;
  children?: string;
}

const defaultProps = {
  checked: false,
};

const Checkbox: React.SFC<Props> = ({onPress, text, checked, children}: Props) =>  {

  const buttonText = text || children || "";

  return (
      <View style={styles.container}>
        <Text style={styles.buttonText}>
          {buttonText}
        </Text>
        <Switch onValueChange={onPress} value={checked}/>
      </View>
    );
};

Checkbox.defaultProps = defaultProps;

export default Checkbox;
