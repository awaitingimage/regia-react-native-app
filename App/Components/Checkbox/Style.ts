import { StyleSheet } from "react-native";
import { ApplicationStyles, Colors, Fonts, Metrics } from "../../Themes/";

export default StyleSheet.create({

  container: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    alignSelf: "stretch",
    paddingHorizontal: Metrics.baseMargin,
    paddingVertical: Metrics.baseMargin,
    marginHorizontal: Metrics.baseMargin * 2,
    marginVertical: Metrics.baseMargin,
    borderRadius: 4,
    borderWidth: 0.5,
    borderColor: Colors.darkgrey,
  },
  buttonText: {
    fontFamily: Fonts.type.bold,
    fontSize: Fonts.size.medium,
    color: Colors.primaryText,
    textAlign: "center",
    paddingRight: Metrics.baseMargin,
  },
});
