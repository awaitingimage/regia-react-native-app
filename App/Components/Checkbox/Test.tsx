/// <reference types="@types/jest" />
import * as React from "react";
import * as renderer from "react-test-renderer";
import Checkbox from "./Checkbox";

test("Checkbox component renders correctly", () => {
  const tree = renderer.create(<Checkbox onPress={() => {}}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
