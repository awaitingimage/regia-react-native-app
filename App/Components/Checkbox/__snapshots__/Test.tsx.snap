// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderColor": "#333333",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": 20,
      "marginVertical": 10,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontFamily": "Raleway-Bold",
        "fontSize": 14,
        "paddingRight": 10,
        "textAlign": "center",
      }
    }
  >
    
  </Text>
  <RCTSwitch
    disabled={false}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        undefined,
      ]
    }
    value={false}
  />
</View>
`;
